pkg/aws/metadata/metadata.go

```diff
@@ -11,53 +11,76 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
 package metadata
 
 import (
+	"context"
 	"fmt"
 	"io"
-	"net/http"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
 )
 
-func getMetadata(name string) (string, error) {
-	resp, err := http.Get("http://169.254.169.254/latest/meta-data/" + name)
+func newClient() (*imds.Client, error) {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+
+	client := imds.NewFromConfig(cfg)
+
+	return client, err
+}
+
+func getMetadata(client *imds.Client, path string) (string, error) {
+	res, err := client.GetMetadata(context.TODO(), &imds.GetMetadataInput{
+		Path: path,
+	})
 	if err != nil {
-		return "", fmt.Errorf("unable to retrieve instance-id from metadata server: %s", err)
+		return "", fmt.Errorf("unable to retrieve AWS metadata %s: %w", path, err)
 	}
 
-	defer resp.Body.Close()
-	instanceID, err := io.ReadAll(resp.Body)
+	defer res.Content.Close()
+	value, err := io.ReadAll(res.Content)
 	if err != nil {
-		return "", fmt.Errorf("unable to read response body: %s", err)
+		return "", fmt.Errorf("unable to read response content for AWS metadata %s: %w", path, err)
```

Comment:
Let's make the `path` a bit easier to read in the error msg:

```suggestion
		return "", fmt.Errorf("unable to read response content for AWS metadata %q: %w", path, err)
```